### PR TITLE
feat: impl teamsfx add cicd

### PR DESCRIPTION
**Pull request by IvanJobs**
to finish: https://dev.azure.com/msazure/Microsoft%20Teams%20Extensibility/_workitems/edit/13882675

1. `teamsfx add cicd` interactively.
2. `teamsfx add cicd --env dev --provider github --template ci cd --interactive false` non-interactively.

E2E TEST: https://github.com/OfficeDev/TeamsFx/blob/838ca43f447155ddd124a187977ae50c27e2d3ca/packages/cli/tests/e2e/cicd/VerifyGeneratedTemplates.tests.ts#L19

### CODE REVIEW

**Review by MuyangAmigo — APPROVED** [review]
CLI Signature and Description Looks Good to Me.

**Comment by wenytang-ms** [comment]
just rebase or merge dev branch, the audit problem will fixed.

**Review by Alive-Fish — DISMISSED** [review]

**Review by Alive-Fish — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/cicd/index.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export class CICDPluginV2 implements ResourcePlugin {
     tokenProvider: TokenProvider
   ): Promise<FxResult> {
     // add CI CD workflows for minimal app is not supported.
-    if (ctx.projectSetting && isPureExistingApp(ctx.projectSetting)) {
+    if (inputs.platform !== Platform.CLI_HELP && isPureExistingApp(ctx.projectSetting)) {
```
Comment (eriolchan) [inline packages/fx-core/src/plugins/resource/cicd/index.ts:72]: What does this if condition mean?

File: packages/cli/src/cmds/add.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    CliTelemetry.withRootFolder(rootFolder).sendTelemetryEvent(TelemetryEvent.AddCICDStart);
+
+    const result = await activate(rootFolder);
+    if (result.isErr()) {
+      CliTelemetry.sendTelemetryErrorEvent(TelemetryEvent.AddCICD, result.error);
+      return err(result.error);
+    }
+    const core = result.value;
+    let inputs: Inputs;
+    {
```
[the remark this answers is not in the document]
Comment (IvanJobs) [reply packages/cli/src/cmds/add.ts:47]: I just copied the pattern from existing code and the same pattern exists in `cmds/validate.ts` and other. 
For me, it seems unnecessary, let me try to remove it next time.

File: packages/fx-core/src/plugins/resource/cicd/index.ts (hunk 1 above)
Comment (IvanJobs) [reply packages/fx-core/src/plugins/resource/cicd/index.ts:72]: The original code is added by @qinezh to give a nice error in minimal app scenario. 
I found a bug (when CLI calls this getQuestions for CLI `--help`, it should not throw this error) when I implementing the CLI, so fix it along the way.
BTW, ctx.projectSetting is not an undefined type, so remove the useless if (ctx.projectSetting).

**Pull request merged**